File: src/main/java/org/example/classroom/enums/UserRole.java
package org.example.classroom.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 用户角色枚举
 * <p>
 * 定义系统中的用户角色类型
 *
 * @author Classroom Team
 * @since 2025-02-23
 */
@Getter
@AllArgsConstructor
public enum UserRole {

    /**
     * 管理员
     */
    ADMIN("admin", "管理员", 1),

    /**
     * 教师
     */
    TEACHER("teacher", "教师", 2),

    /**
     * 学生
     */
    STUDENT("student", "学生", 3);

    /**
     * 角色编码（存储在数据库）
     */
    private final String code;

    /**
     * 角色描述
     */
    private final String description;

    /**
     * 角色级别（数字越小权限越高）
     */
    private final int level;

    /**
     * 根据编码获取角色枚举
     *
     * @param code 角色编码
     * @return 角色枚举
     * @throws IllegalArgumentException 如果编码无效
     */
    public static UserRole fromCode(String code) {
        if (code == null) {
            return null;
        }
        for (UserRole role : values()) {
            if (role.code.equals(code)) {
                return role;
            }
        }
        throw new IllegalArgumentException("Invalid role code: " + code);
    }

    /**
     * 判断是否为管理员
     */
    public boolean isAdmin() {
        return this == ADMIN;
    }

    /**
     * 判断是否为教师
     */
    public boolean isTeacher() {
        return this == TEACHER;
    }

    /**
     * 判断是否为学生
     */
    public boolean isStudent() {
        return this == STUDENT;
    }

    /**
     * 判断权限是否高于指定角色
     *
     * @param other 另一个角色
     * @return true 如果当前角色级别更高
     */
    public boolean hasHigherLevelThan(UserRole other) {
        return this.level < other.level;
    }
}

File: src/main/resources/static/classroom/src/enums/UserRole.js
/**
 * 用户角色枚举
 * 与后端 UserRole.java 保持一致
 *
 * @author Classroom Team
 * @since 2025-02-23
 */

export const UserRole = {
	ADMIN: 'ADMIN',
	TEACHER: 'TEACHER',
	STUDENT: 'STUDENT',
};

/**
 * 角色显示名称映射
 */
export const UserRoleLabels = {
	[UserRole.ADMIN]: '管理员',
	[UserRole.TEACHER]: '教师',
	[UserRole.STUDENT]: '学生',
};

/**
 * 角色首页路径映射
 */
export const UserRoleHomePaths = {
	[UserRole.ADMIN]: '/adminDashboard',
	[UserRole.TEACHER]: '/teacherDashboard',
	[UserRole.STUDENT]: '/studentDashboard',
};

/**
 * 根据角色获取显示名称
 * @param {string} role - 角色值
 * @returns {string} 显示名称
 */
export function getUserRoleLabel(role) {
	return UserRoleLabels[role] || '未知角色';
}

/**
 * 根据角色获取首页路径
 * @param {string} role - 角色值
 * @returns {string} 首页路径
 */
export function getUserRoleHomePath(role) {
	return UserRoleHomePaths[role] || '/login';
}

export default {
	UserRole,
	UserRoleLabels,
	UserRoleHomePaths,
	getUserRoleLabel,
	getUserRoleHomePath,
};
